File: app/src/main/java/com/kite/mnemoai/ui/vocabularybook/VocabularyUIState.java
package com.kite.mnemoai.ui.vocabularybook;

import com.kite.mnemoai.data.model.DailyStatistic;

import java.util.List;

public class VocabularyUIState {
    private final List<LearningVocabularyBookItem> learningVocabularyBookItems;
    private final List<AllVocabularyBookItem> allVocabularyBookItems;
    private final int newLearningWordCount;
    private final DailyStatistic dailyStatistic;

    public VocabularyUIState(List<LearningVocabularyBookItem> learningVocabularyBookItems, List<AllVocabularyBookItem> allVocabularyBookItems, int newLearningWordCount, DailyStatistic dailyStatistic) {
        this.learningVocabularyBookItems = learningVocabularyBookItems;
        this.allVocabularyBookItems = allVocabularyBookItems;
        this.newLearningWordCount = newLearningWordCount;
        this.dailyStatistic = dailyStatistic;
    }


    public List<LearningVocabularyBookItem> getLearningVocabularyBookItems() {
        return learningVocabularyBookItems;
    }

    public List<AllVocabularyBookItem> getAllVocabularyBookItems() {
        return allVocabularyBookItems;
    }

    public int getNewLearningWordCount() {
        return newLearningWordCount;
    }

    public DailyStatistic getDailyStatistic() {
        return dailyStatistic;
    }
}

File: feature/vocabularybook/src/main/java/com/kite/mnemoai/vocabularybook/VocabularyUIState.kt
package com.kite.mnemoai.vocabularybook

import com.kite.mnemoai.model.group.AllVocabularyBookItem
import com.kite.mnemoai.model.group.LearningVocabularyBookItem
import com.kite.mnemoai.model.statistic.DailyStatistic

data class VocabularyUIState(
    val learningVocabularyBookItems: List<LearningVocabularyBookItem>?,
    val allVocabularyBookItems: List<AllVocabularyBookItem>?,
    val newLearningWordCount: Int,
    val dailyStatistic: DailyStatistic?,
)
